src/backend/distributed/relay/relay_event_utility.c

```diff
@@ -473,6 +476,80 @@ RelayEventExtendNamesForInterShardCommands(Node *parseTree, uint64 leftShardId,
 
 
 /*
+ * AlterSequenceMinMaxValueCommand constructs an ALTER SEQUENCe statmenet which sets
+ * the start, restart, minvalue and maxvalue of sequence with the given name and schema.
+ * The function provides the uniqueness by shifting the start of the sequence by
+ * GetLocalGroupId() << 48 + 1 and sets a maxvalue which stops it from passing out any values
+ * greater than: (GetLocalGroupID() + 1) << 48.
+ *
+ * This is to ensure every group of workers passes out values from a unique range,
+ * and therefore that all values generated for the sequence are globally unique.
+ */
+AlterSeqStmt *
+AlterSequenceMinMaxValueCommand(char *schemaName, char *sequenceName)
+{
+	StringInfo startNumericString = makeStringInfo();
+	StringInfo maxNumericString = makeStringInfo();
+	int64 startValue = 0;
+	int64 maxValue = 0;
+	Node *startFloatArg = NULL;
+	Node *maxFloatArg = NULL;
+	AlterSeqStmt *alterSequenceStatement = makeNode(AlterSeqStmt);
+
+	alterSequenceStatement->sequence = makeRangeVar(schemaName, sequenceName, -1);
+
+	/*
+	 * DefElem->arg can only hold literal ints up to int4, in order to represent
+	 * larger numbers we need to construct a float represented as a string.
+	 */
+	startValue = (((int64) GetLocalGroupId()) << 48) + 1;
+	appendStringInfo(startNumericString, "%lu", startValue);
+	startFloatArg = (Node *) makeFloat(startNumericString->data);
+
+	maxValue = startValue + ((int64) 1 << 48);
+	appendStringInfo(maxNumericString, "%lu", maxValue);
+	maxFloatArg = (Node *) makeFloat(maxNumericString->data);
+
+	SetDefElemArg(alterSequenceStatement, "start", startFloatArg);
+	SetDefElemArg(alterSequenceStatement, "restart", startFloatArg);
```

Comment:
I have updated the part so that it restarts only if necessary. Note that since we call nextval to get the most recent value of the sequence, each `start_metadata_sync_to_node` will increment the sequence by one, but I think it is not a deal breaker.